python/sanlock.c

```
@@ -1325,7 +1336,7 @@ py_killpath(PyObject *self __unused, PyObject *args, PyObject *keywds)
         size_t arg_len;
 
         item = PyList_GetItem(argslist, i);
-        p = PyString_AsString(item);
+        p = pystring_as_cstring(item);
 
         if (p == NULL) {
             __set_exception(EINVAL, "Killpath argument not a string");
```

Same issue - if p is NULL, we already set TypeError, and the SanlockException raised is hides the real error.

We have 2 instances of this issue, lets fix both of them later.

---

one less, one left.

---

We can eliminate this change by squashing my patch rewriting this function. I'm not fan of keeping temporary changes in the tree.

---

We can resolve both by squashing later patches, but if you don't find an easy way we can leave this as is.

---

later patches only covers the use-case  in __parse_resource out of the three occurrences shown here. so we can leave it as is at this point.

---

Merged into my py3 branch.